iib/workers/tasks/build_containerized_merge.py

```diff
@@ -0,0 +1,491 @@
+            self.bundles_queue.task_done()
+
+
+@app.task
+@request_logger
+@instrument_tracing(
+    span_name="workers.tasks.build.handle_containerized_merge_request",
+    attributes=get_binary_versions(),
+)
+def handle_containerized_merge_request(
+    source_from_index: str,
+    deprecation_list: List[str],
+    request_id: int,
+    binary_image: Optional[str] = None,
+    target_index: Optional[str] = None,
+    overwrite_target_index: bool = False,
+    overwrite_target_index_token: Optional[str] = None,
+    distribution_scope: Optional[str] = None,
+    binary_image_config: Optional[str] = None,
+    build_tags: Optional[List[str]] = None,
+    graph_update_mode: Optional[str] = None,
+    ignore_bundle_ocp_version: Optional[bool] = False,
+    index_to_gitlab_push_map: Optional[Dict[str, str]] = None,
+    parallel_threads: int = 5,
+) -> None:
+    """
+    Coordinate the work needed to merge old (N) index image with new (N+1) index image.
+
+    :param str source_from_index: pull specification to be used as the base for building the new
+        index image.
+    :param str target_index: pull specification of content stage index image for the
+        corresponding target index image.
+    :param list deprecation_list: list of deprecated bundles for the target index image.
+    :param int request_id: the ID of the IIB build request.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param bool overwrite_target_index: if True, overwrite the input ``target_index`` with
+        the built index image.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``target_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str distribution_scope: the scope for distribution of the index image, defaults to
+        ``None``.
+    :param build_tags: list of extra tag to use for intermetdiate index image
+    :param str graph_update_mode: Graph update mode that defines how channel graphs are updated
+        in the index.
+    :param bool ignore_bundle_ocp_version: When set to `true` and image set as target_index is
+        listed in `iib_no_ocp_label_allow_list` config then bundles without
+        "com.redhat.openshift.versions" label set will be added in the result `index_image`.
+    :raises IIBError: if the index image merge fails.
+    :param dict index_to_gitlab_push_map: the dict mapping index images (keys) to GitLab repos
+        (values) in order to push their catalogs into GitLab.
+    :param int parallel_threads: the number of parallel threads to use for validating the bundles
+    :raises IIBError: if the index image merge fails.
+    """
+    reset_docker_config()
+    set_request_state(request_id, 'in_progress', 'Preparing request for merge')
+
+    # Prepare request
+    prebuild_info = prepare_request_for_build(
```

Comment:
The non-containerized version has this function call under `with set_registry_token(overwrite_target_index_token, target_index, append=True):` context. Maybe keep it?